kan/spline.py

```diff
@@ -129,10 +130,11 @@ def curve2coef(x_eval, y_eval, grid, k, device="cpu"):
     >>> x_eval = torch.normal(0,1,size=(num_spline, num_sample))
     >>> y_eval = torch.normal(0,1,size=(num_spline, num_sample))
     >>> grids = torch.einsum('i,j->ij', torch.ones(num_spline,), torch.linspace(-1,1,steps=num_grid_interval+1))
-    >>> curve2coef(x_eval, y_eval, grids, k=k).shape
     torch.Size([5, 13])
     '''
     # x_eval: (size, batch); y_eval: (size, batch); grid: (size, grid); k: scalar
-    mat = B_batch(x_eval, grid, k, device=device).permute(0, 2, 1).to(y_eval.dtype)
-    coef = torch.linalg.lstsq(mat.to('cpu'), y_eval.unsqueeze(dim=2).to('cpu')).solution[:, :, 0]  # sometimes 'cuda' version may diverge
+    mat = B_batch(x_eval, grid, k, device=device).permute(0, 2, 1)
```

Comment:
Could you please confirm if it is compatible with float64? Please check the context of the changes made in PR #148.

Reply:
It works with float64 inputs. I agree with Ziming's reply to #146. The singular problem matters. In fact, _torch.linalg.lstsq_ may raise errors due to a singular problem using _gels_ as the driver in cuda version because _gels_ can only work with a full-rank **mat**. 

My other concern about this modification is that I just noticed the author commented the original code `# sometimes 'cuda' version may diverge` . I don't know if this modification would raise the 'diverge' situation again while running on 'cuda'. For now, this modification at least fixed the _Runtime Error_.

Reply:
I actually tested it thoroughly through all singularity problems I could think of. Of course I don't expect this to work in all situations, but at least now it's working on **most** :)
We'll have to understand better when a non full-rank mat actually appears and how to mitigate the issue - from a theoretic point of view, I can only see this kind of problem in a feature being totally excluded by the graph, which indeed would be a valuable information by itself.